docker/maistra-builder_3.0.Dockerfile

```diff
@@ -15,6 +15,12 @@ ENV CONTAINERD_VERSION=1.7.22
 ENV DOCKER_BUILDX_VERSION=0.17.1
 ENV K8S_TEST_INFRA_VERSION=4f5d74517ae8ec387568f8c766f4b0ba1f454129
 
+# install rust >= 1.85 needed for ztunnel
+RUN curl https://sh.rustup.rs -sSf > installer; \
+    sh installer -y; \
+    export PATH=/root/.cargo/bin:$PATH; \
+    cargo --version
```

Comment:
nit: do we need this?

nit: we should remove `installer.sh`, or run it with a pipe like https://github.com/maistra/test-infra/blob/main/docker/maistra-builder_2.6.Dockerfile#L464-L467